components/org.wso2.micro.integrator.core/src/main/java/org/wso2/micro/service/mgt/ServiceAdmin.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.micro.service.mgt;
+
+import org.apache.axis2.AxisFault;
+import org.apache.axis2.description.AxisOperation;
+import org.apache.axis2.description.AxisService;
+import org.apache.axis2.description.AxisServiceGroup;
+import org.apache.axis2.description.Parameter;
+import org.apache.axis2.engine.AxisConfiguration;
+import org.wso2.carbon.service.mgt.util.Utils;
+import org.wso2.carbon.utils.deployment.GhostDeployerUtils;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+public class ServiceAdmin {
+
+    private static AxisConfiguration axisConfiguration = null;
+    private static ServiceAdmin serviceAdmin = null;
+
+    private ServiceAdmin() {
+
+    }
+
+    public static ServiceAdmin getInstance() {
+
+        if (serviceAdmin == null) {
+            return new ServiceAdmin();
+        }
+        return serviceAdmin;
+    }
+
+    public void init(AxisConfiguration axisConfiguration) {
+
+        ServiceAdmin.axisConfiguration = axisConfiguration;
+    }
+
+    public Boolean isInitailized() {
+
+        if (axisConfiguration != null) {
+            return true;
+        }
+        return false;
+    }
+
+    public ServiceMetaData getServiceData(String serviceName) throws Exception {
+
+        AxisService service = axisConfiguration.getServiceForActivation(serviceName);
+        if (service == null) {
+            Map<String, AxisService> transitGhostServicesMap = GhostDeployerUtils.getTransitGhostServicesMap(axisConfiguration);
+            service = transitGhostServicesMap.get(serviceName);
+        }
+
+        if (service == null) {
+            String msg = "Invalid service name, service not found : " + serviceName;
+            throw new AxisFault(msg);
+        }
+
+        // if the existing service is a ghost service, deploy the actual one
+        if (GhostDeployerUtils.isGhostService(service)) {
```

Comment:
We have removed this . We can skip this .

Reply:
+1. done